src/plotter_disk.hpp

```diff
@@ -73,8 +76,639 @@ struct Phase3Results {
     uint32_t header_size;
 };
 
+static void print_buf(const unsigned char* buf, size_t buf_len)
+{
+    size_t i = 0;
+    for (i = 0; i < buf_len; ++i)
+        fprintf(stdout, "%02X%s", buf[i], (i + 1) % 16 == 0 ? "\r\n" : " ");
+}
+
 const Bits empty_bits;
 
+#define STRIPESIZE 8192
+#define NUMTHREADS 2
+
+typedef struct {
```

Comment:
in C++ you don't need to create typedefs for classes, their names are proper types. You can just do:

```
struct THREADDATA
{
   // ...
};
```